### ISSUE TITLE
Built-in analyzers are used when EnableNETAnalyzers is true and the Microsoft.CodeAnalysis.NetAnalyzers NuGet package is installed

### ISSUE THREAD

**Issue by reduckted** [dotnet/roslyn-analyzers#4849]
### Analyzer

N/A
### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3

### Describe the bug

If you set `EnableNETAnalyzers` to true _and_ install the `Microsoft.CodeAnalysis.NetAnalyzers` NuGet package, the build will report these warnings:

in VB.NET:
```
vbc : warning BC42378: Unable to load analyzer assembly D:\NuGet\packages\microsoft.codeanalysis.netanalyzers\5.0.3\analyzers\dotnet\vb\Microsoft.CodeAnalysis.NetAnalyzers.dll : Assembly with same name is already loaded.
vbc : warning BC42378: Unable to load analyzer assembly D:\NuGet\packages\microsoft.codeanalysis.netanalyzers\5.0.3\analyzers\dotnet\vb\Microsoft.CodeAnalysis.VisualBasic.NetAnalyzers.dll : Assembly with same name is already loaded.
```

or in C#:
```
CSC : warning CS8034: Unable to load Analyzer assembly D:\NuGet\packages\microsoft.codeanalysis.netanalyzers\5.0.3\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll : Assembly with same name is already loaded
CSC : warning CS8034: Unable to load Analyzer assembly D:\NuGet\packages\microsoft.codeanalysis.netanalyzers\5.0.3\analyzers\dotnet\cs\Microsoft.CodeAnalysis.NetAnalyzers.dll : Assembly with same name is already loaded
```

### Steps To Reproduce

Use this project file:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

### Expected behavior

Installing the NuGet package should completely turn off the built-in analyzers according to [the docs](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#latest-updates):

> Install the `Microsoft.CodeAnalysis.NetAnalyzers` NuGet package to decouple rule updates from .NET SDK updates. **Installing the package turns off the built-in SDK analyzers and generates a build warning if the SDK contains a newer analyzer assembly version than that of the NuGet package.**

### Actual behavior

Warnings are reported, and the analyzers that come from the SDK are used, even though the analyzers from the NuGet packages have a higher version number.

**Comment by reduckted** [comment]
I'm not sure if this is the correct repository for this issue, but hopefully it can be transferred elsewhere if needed.

Am I correct in assuming that if you install the NuGet package, then `EnableNETAnalyzers` doesn't need to be specified? I have some projects that need to target `netstandard2.0` and others that target `net5.0` in the same solution, so I turned on `EnableNETAnalyzers` in a `Directory.Build.props` file, but due to #4840 I've also had to install the NuGet package. That led me to seeing these warnings, as well as the warnings from #4840 because the built-in analyzers are being used.

**Comment by Youssef1313** [comment]
I think when using the NuGet package, you don't need to set `EnableNETAnalyzers` at all (neither to true nor false).

The docs may be updated to be more clear about that.

Also tagging @mavasani to confirm, and @gewarren for doc update if needed.

**Comment by mavasani** [comment]
> Built-in analyzers are used when EnableNETAnalyzers is true and the Microsoft.CodeAnalysis.NetAnalyzers NuGet package is installed

This is by design. `EnableNETAnalyzers` is to enable the built-in analyzers and should not be used when installing the NuGet package. We should update the docs to clarify this. I'll port it to docs repo for @gewarren
